rio/src/RIO/Prelude/Logger.hs

```diff
@@ -287,41 +288,18 @@ logOptionsHandle
   -> Bool -- ^ verbose?
   -> m LogOptions
 logOptionsHandle handle' verbose = liftIO $ do
-  terminal <- hIsTerminalDevice handle'
   useUtf8 <- canUseUtf8 handle'
-  unicode <- if useUtf8 then return True else getCanUseUnicode
+  terminal <- hIsTerminalDevice handle'
   return LogOptions
     { logMinLevel = if verbose then LevelDebug else LevelInfo
     , logVerboseFormat = verbose
     , logTerminal = terminal
     , logUseTime = verbose
     , logUseColor = verbose && terminal
-    , logSend = \builder ->
-        if useUtf8 && unicode
-          then hPutBuilder handle' (builder <> flush)
-          else do
-            let lbs = toLazyByteString builder
-                bs = toStrictBytes lbs
-            case decodeUtf8' bs of
-              Left e -> error $ "mkLogOptions: invalid UTF8 sequence: " ++ show (e, bs)
-              Right text -> do
-                let text'
-                      | unicode = text
-                      | otherwise = T.map replaceUnicode text
-                TIO.hPutStr handle' text'
```

Comment:
This branch is definitely _not_ Stack specific. In the new version of the code, you seem to be ignoring the `logUseUtf8` value entirely. This code will not work on any machine with a locale/code page setting that does not support UTF-8.

Reply:
If you take a look at the replaceUnicode function, it only replaces two codepoints specific to GHC output.

Yes, `logUseUtf8` is ignored entirely, it's intended to be read by the user.  On further review, that is probably unexpected and probably should have an accompanying `askUseUtf8` or something.

So, what should we do here?  There are a few different ways to munge unicode into 7 bit ASCII

Reply:
Do other loggers just punt on this?  Why is it particularly coming up here?  Just that stack has been used a bunch on a variety of platforms?

Reply:
I think you're missing the exact line of code that I'm commenting on. The way the code works today is that, if the file handle doesn't support UTF8, we decode the bytestring to Text and then use the textual I/O functions to output it.

Reply:
I see.  I'm not familiar with how `hPutBuilder handle' (builder <> flush)` would differ from `T.hPutStrLn (decodeUtf8 ...)`.  Seems like it should be pretty equivalent.

Reply:
Ah, I guess it's because in theory it will work for things that aren't utf-8.  Having a hard time finding where in the code that happens, though.  I guess the stuff in System.IO does that.  It looks like the text io stuff will have different concurrent behavior, potentially interspersing output of chars.